src/group_theory/monoid_localization.lean

```diff
@@ -147,37 +158,52 @@ namespace localization_map
 
 variables {S N}
 
-@[to_additive, ext] lemma ext {f g : localization_map S N} (h : ∀ x, f.1 x = g.1 x) : f = g :=
-by cases f; cases g; simp only []; exact monoid_hom.ext h
+/-- Makes a localization map from a `comm_monoid` hom satisfying the characteristic predicate. -/
+@[to_additive "Makes a localization map from an `add_comm_monoid` hom satisfying the characteristic predicate."]
+def of_monoid_hom (f : M →* N) (H1 : ∀ y : S, is_unit (f y))
+  (H2 : ∀ z, ∃ x : M × S, z * f x.2 = f x.1) (H3 : ∀ x y, f x = f y ↔ ∃ c : S, x * c = y * c) :
+  localization_map S N :=
+⟨f, f.map_one, f.map_mul, H1, H2, H3⟩
```

Comment:
Here you could write
```
{ map_units' := H1,
  surj' := H2,
  eq_iff_exists' := H3,
  .. f }
```
I think this is better, for some technical reasons that I don't actually know off the top of my head.